### PR TITLE
Add `EditorSidebar` component

### PR DESCRIPTION
**Pull request by peppy**
As per new designs. Will eventually replace the floating toolbox displays we currently use. Intending on using this immediately in the skin editor to clean up its janky interface.

osu Game Tests 2022-03-15 at 07 39 03

Note that some metrics were done via eyeballing. There's some disconnect between what can be gained from figma due to the differences in how text height is calculated.

### CODE REVIEW

File: osu.Game/Screens/Edit/Components/EditorSidebarSection.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            [BackgroundDependencyLoader]
+            private void load(OverlayColourProvider colourProvider)
+            {
+                InternalChildren = new Drawable[]
+                {
+                    new OsuSpriteText
+                    {
+                        Text = text,
+                        Font = OsuFont.Default.With(size: 16, weight: FontWeight.SemiBold),
+                    },
```
Comment (bdach) [inline osu.Game/Screens/Edit/Components/EditorSidebarSection.cs:64]: This implementation is not super resilient against text potentially not fitting in a single line which may become a problem especially with localisation. Maybe fine to ignore until it actually becomes a problem, but shouldn't take too much work to address now either if you decide it's worth it.

**Review by bdach — COMMENTED** [review]
Generally looks good, just one passing remark that may not require action just yet

File: osu.Game/Screens/Edit/Components/EditorSidebarSection.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Screens/Edit/Components/EditorSidebarSection.cs:64]: Fair point, I'll make it wrap (thought I'd probably hope we don't do this as it won't look amazing).

Comment (smoogipoo) [reply osu.Game/Screens/Edit/Components/EditorSidebarSection.cs:64]: Probably best to address if it's a simple change (seems like just using TFC).

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**